www/guides/react.md

```diff
@@ -0,0 +1,373 @@
+      </header>
+    </div>
+  );
+}
+
+export default App;
+```
+
+Now include it in `index.js`
+
+```diff
+import React from 'react';
+import ReactDOM from 'react-dom';
++ import App from './App';
+
+- ReactDOM.render(<div>"HELLO WORLD"</div>, document.getElementById('root'));
++ ReactDOM.render(
++    <React.StrictMode>
++      <App />
++    </React.StrictMode>,
++    document.getElementById('root'),
++  );
+```
+
+> [Strict Mode](https://reactjs.org/docs/strict-mode.html) is a tool for highlighting potential problems in React code.
+
+You shouldn't need to restart Snowpack to see this, it should look like this:
+
+IMAGE: screenshot of the project, showing text saying "edit `src/App.js` and save to reload," text that says "Page has been open for" and the number of seconds then "seconds". A link with the text Learn React that links to the React website.
+
+## Styling your Snowpack/React project
+
+When you add assets like images or CSS, Snowpack includes them in your final build. If you already know React, this process should look pretty familiar.
+
+> As you're doing this, you should not need to reload the page or restart Snowpack. Snowpack updates the project in the browser as you edit code, which is pretty nifty.
+
+Add this file `logo.svg` (link to file) to `src`. Now you can import it into your `App.jsx` and use it in an `img` tag to display it.
+
+```diff
+import React, { useState, useEffect } from 'react';
++ import logo from './logo.svg';
+
+function App() {
+  // Create the count state.
+  const [count, setCount] = useState(0);
+  // Create the counter (+1 every second).
+  useEffect(() => {
+    const timer = setTimeout(() => setCount(count + 1), 1000);
+    return () => clearTimeout(timer);
+  }, [count, setCount]);
+  // Return the App component.
+  return (
+    <div className="App">
+      <header className="App-header">
++        <img src={logo} className="App-logo" alt="logo" />
+```
+
+Image: the React logo (a blue atom) is now at the top of the page
+
+The project already has index.css for global styles, but component CSS can help keep code organized. Snowpack knows to only load this CSS when displaying the `App` component, which means it's not loading unnecessary CSS.
+
+Create `src/App.css` and add this CSS:
+
+```css
+.App {
+  text-align: center;
+}
+.App code {
+  background: #fff3;
+  padding: 4px 8px;
+  border-radius: 4px;
+}
+.App p {
+  margin: 0.4rem;
+}
+.App-logo {
+  height: 40vmin;
+  pointer-events: none;
+}
+
+@media (prefers-reduced-motion: no-preference) {
+  .App-logo {
+    animation: App-logo-spin infinite 20s linear;
+  }
+}
+
+.App-header {
+  background-color: #282c34;
+  min-height: 100vh;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  font-size: calc(10px + 2vmin);
+  color: white;
+}
+
+.App-link {
+  color: #61dafb;
+}
+
+@keyframes App-logo-spin {
+  from {
+    transform: rotate(0deg);
+  }
+  to {
+    transform: rotate(360deg);
+  }
+}
```

Comment:
what's the bare minimum to get the logo to spin and things to be centered okay-ishly, this is a lot of scrolling lol